repo/repo.go

```diff
@@ -100,6 +107,55 @@ func ImportRepoRules(filename string, repoCache *RemoteCache) ([]*rule.Rule, err
 	return rules, nil
 }
 
+// MergeRules merges a list of generated repo rules with the already defined repo rules,
+// and then updates each rule's underlying file. If the generated rule matches an existing
+// one, then it inherits the file where the existing rule was defined. If the rule is new then
+// its file is set as the destFile parameter. A list of the updated files is returned.
+func MergeRules(genRules []*rule.Rule, existingRules map[*rule.File][]string, destFile *rule.File, kinds map[string]rule.KindInfo) []*rule.File {
+	sort.Stable(byRuleName(genRules))
+
+	rulesByFile := make(map[*rule.File][]*rule.Rule)
+	for file, repoNames := range existingRules {
```

Comment:
I'm having trouble understanding how this loop works.

It's building `rulesByFile`, so for each generated rule, it needs to check whether there is a file in `existingRules`: if so, the generated rule should go there; otherwise, it should go in `destFile`.

I think it would be simpler to build a `map[string]*rule.File` from `existingRules`. Then building `rulesByFile` would be simpler.